llpc/lower/llpcSpirvLowerConstImmediateStore.cpp

```diff
@@ -223,16 +223,12 @@ bool SpirvLowerConstImmediateStore::tryProcessAlloca(AllocaInst *allocaInst) {
   }
 
   // Step 3: Create the global variable and replace the alloca
-  GlobalVariable *global = nullptr;
-  auto iter = m_allocToGlobals.find(initializer);
-  if (iter == m_allocToGlobals.end()) {
+  GlobalVariable *&global = m_allocToGlobals[initializer];
+  if (!global) {
```

Comment:
This assumes that if the key `initializer` was not found in `m_allocToGlobals`, a new entry is created and the value of the new entry is `nullptr`, right?

`DensedMap::InsertIntoBucket` creates the new value with `::new (&TheBucket->getSecond()) ValueT(std::forward<ValueArgs>(Values)...);` (where `Values` is empty), so it calls the default constructor of `GlobalVariable*`.
If I am not mistaken, the default constructor of a pointer **does not initialize memory**, so checking for `nullptr` afterwards only works when we are lucky.

`try_emplace(initializer, nullptr)` should do the correct thing.

Reply:
> But as DenseMap doesn't explicitly document any of this, using it conservatively won't hurt.

My feeling is that DenseMap behaves exactly like std::map in this regard, so it's fine to rely on the value initialization.

Reply:
That value initialization is a good point. This code looks like it does the correct thing then :)